### PR TITLE
Clearing an alarm causes an EOL error

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

The `tedge-to-te` topic converter was not properly handling the `alarm` clear message. This PR addresses this issue.

Also, added the tests that cover all the missing topic conversion cases like empty messages, empty topics, Unsupported topics, etc.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
https://github.com/thin-edge/thin-edge.io/issues/2151

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs [hunk 1]
```diff
@@ -85,10 +85,15 @@ async fn convert_incoming_main_device_alarm_topic() -> Result<(), DynError> {
...
-    (left.topic == right.topic) && (left_msg == right_msg)
+    if !left.payload_str().unwrap().is_empty() && !right.payload_str().unwrap().is_empty() {
+        let left_msg: serde_json::Value = serde_json::from_slice(left.payload.as_bytes()).unwrap();
+        let right_msg: serde_json::Value =
+            serde_json::from_slice(right.payload.as_bytes()).unwrap();
+
+        (left.topic == right.topic) && (left_msg == right_msg)
+    } else {
+        left.topic == right.topic
+    }
```
[the remark this answers is not in the document]
Comment (didier-wenzek) [reply crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs:96]: Instead of unwrapping the errors, one can convert these to options. This would keep the code simple.

```
    let left_msg: serde_json::Value = serde_json::from_slice(left.payload.as_bytes()).ok();
    let right_msg: serde_json::Value = serde_json::from_slice(right.payload.as_bytes()).ok();
    (left.topic == right.topic) && (left_msg == right_msg)
```

[the remark this answers is not in the document]
Comment (PradeepKiruvale) [reply crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs]: Updated,

File: crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs [hunk 2]
```diff
@@ -119,6 +124,117 @@ async fn convert_incoming_custom_main_device_alarm_topic() -> Result<(), DynErro
...
+
+    // Simulate SoftwareList MQTT message received.
+    let mqtt_message = MqttMessage::new(&Topic::new_unchecked("tedge///MyCustomAlarm"), "");
+
+    mqtt_box.send(mqtt_message).await?;
+
+    // Assert SoftwareListRequest
+    mqtt_box
+        .assert_received_matching(same_json_over_mqtt_msg, [])
+        .await;
```
[the remark this answers is not in the document]
Comment (PradeepKiruvale) [reply crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs]: moved to converter.rs, now tests the converter rather than the actor.

**Review by didier-wenzek — APPROVED** [review]
Approved
